src/process.c

```diff
@@ -916,11 +917,24 @@ cc_oci_vm_launch (struct cc_oci_config *config)
 		g_debug ("reading hypervisor command-line length from pipe");
 		bytes = read (hypervisor_args_pipe[0], &hypervisor_args_len,
 			sizeof (hypervisor_args_len));
-		if (bytes < 0) {
+		if (bytes < 0 || hypervisor_args_len < 0) {
 			g_critical ("failed to read hypervisor args length");
 			goto child_failed;
 		}
 
+		/* Perform a basic validation check.
+		 *
+		 * ARG_MAX is technically the maximum size of the args *and*
+		 * the environment for a process, but atleast this
+		 * provides an upper-bound to protect against any
+		 * potential DoS.
+		 */
+		if (hypervisor_args_len > ARG_MAX) {
+			g_critical ("max args len is %d, but parent sent %d",
+					ARG_MAX, hypervisor_args_len);
+			goto child_failed;
+		}
+
 		hypervisor_args = g_new0(gchar, (gsize)(1+hypervisor_args_len));
```

Comment:
Lets assume that ```hypervisor_args_len``` is equal to ```ARG_MAX``` , at this point we are allocating ARG_MAX+1.
Probably you should modify ```if (hypervisor_args_len > ARG_MAX)``` with ```if (hypervisor_args_len >= ARG_MAX)```